docs/adrs/0077-permanently-store-origin-duty-gbloc.md

```diff
@@ -0,0 +1,107 @@
+---
+title: 0077 Permanently store Origin Duty Location GBLOC
+description: |
+  Decision outcome: Add a column to the orders table for Origin Duty Location GBLOC
+---
+
+# Permanently Store the Origin Duty Location GBLOC
+
+**User Story:** *[MB-15032](https://dp3.atlassian.net/browse/MB-15032)* :lock:
+
+## Problem Statement
+
+In our current database structure, GBLOCs are stored in the `postal_code_to_gblocs` as a one-to-many value: one GBLOC can represent many postal codes. Move origin duty locations associate to GBLOCs using the `origin_duty_location_to_gbloc` view. This view queries the `postal_code_to_gblocs` table on every read to fetch the GBLOC associated with the origin postal code on the orders while having the benefit of behaving like a table that can be joined in other queries.
+
+The downside of using a view to fetch the origin duty location GBLOC is that the data remains dynamic, and there is no way to determine what the *original* GBLOC was if the postal code to GBLOC relationship changes later. Since GBLOCs can be consolidated, USTC needs to be able to access the historical data for moves. 
+
+### Additional considerations behind this change
+
+* View tables are not transferred to data warehouse. USTC needs the duty location GBLOC data warehoused.
+* The `origin_duty_location_to_gbloc` does not handle GBLOC outliers, such as USMC logic, leading to [incorrect data](https://dp3.atlassian.net/browse/MB-15143) :lock:.
+
+### Important distinction to note
+In addition to `origin_duty_location_to_gbloc`, there is another view for `move_to_gbloc`. This is used to fetch the Shipment GBLOC, which is based
+on the origin address of the first shipment in a move. The Shipment GBLOC and Origin Duty Location GBLOC are both fetched, and the API manages the logic
+for which GBLOC to select to route the move to the correct queue. 
+
+For the purpose of the current requirement, the `move_to_gbloc` view will remain in place, and the routing logic in the API will not be changed. The primary change
+will be in storing the origin duty location GBLOC when an MTO is created, and query the stored origin duty location GBLOC.
```

Comment:
Thanks for adding a primer around some of the related tables and views. This is a really helpful distinction between the gbloc associated with the origin duty location and the gbloc associated with the move/shipment.